### PR TITLE
feat: implement dlc-btc-lib ethereum handler and ethereum functions

### PR DESCRIPTION
**Pull request by Polybius93**
This pull request removes the existing ethereum contract handlings and hooks, replacing them with the ethereum functions and handler interface from ``dlc-btc-lib``.

### CODE REVIEW

File: src/app/functions/ethereum-account.functions.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  alert(
+    'Your current Ethereum provider is not MetaMask. Please ensure that MetaMask is your active Ethereum wallet and disable any other Ethereum wallets in your browser, then reload the page.'
+  );
+  throw new Error('Non-MetaMask Ethereum provider');
+}
+
+function validateMetaMask(provider: any): boolean {
+  return (
+    typeof provider === 'object' &&
+    provider !== null &&
```
Comment (sosaucily) [inline src/app/functions/ethereum-account.functions.ts:21]: are these related to this feature of using the dlc-btc-lib? seems odd that this is here, as it doesn't seem related to bringing in the lib. Or is it moved from another place?

File: contracts.devnet.json
Comment (Polybius93) [inline contracts.devnet.json:1]: Previously ``deploymentPlans`` were fetched from our **GitHub** ``dlc-solidity`` repository. 
We are expecting that ``deploymentPlans`` won't be changing that frequently anymore, so we can store it in the repository instead of fetching. That would spare us some refetching at rerenders.

File: src/app/hooks/use-ethereum.ts
Comment (Polybius93) [inline src/app/hooks/use-ethereum.ts:1]: The ``useEthereum()`` hook was removed as its functions are now redundant. The ``dlc-btc-lib`` updates include all the necessary functions and more.

File: src/app/hooks/use-ethereum-observer.ts
Comment (Polybius93) [inline src/app/hooks/use-ethereum-observer.ts:1]: Since we no longer have the ``useEthereumContext()`` and its source, ``useEthereumContracts()``, we now use the ``EthereumNetworkConfigurationContext``, which includes a function to return the required ``readOnlyDLCManagerContract``. We no longer need to monitor the changing state of the ``EthereumContext`` or check if ``observerDLCManagerContract`` is *nil*; we simply instantiate it when the function is called.

File: src/app/hooks/use-ethereum-context.ts
Comment (Polybius93) [inline src/app/hooks/use-ethereum-context.ts:1]: The ``useEthereumContext()`` hook was removed as ``EthereumContext`` was replaced by ``EthereumNetworkConfigurationContext``

File: src/app/hooks/use-ethereum-account.ts
Comment (Polybius93) [inline src/app/hooks/use-ethereum-account.ts:1]: The ``useEthereumAccount()`` hook has been removed, with its contents transferred to ``ethereum-account.functions.ts``. This adjustment was made because these functions no longer needed to depend on the dynamic values that ``useEthereumAccount()`` provided.

File: src/app/functions/ethereum.functions.ts
Comment (Polybius93) [inline src/app/functions/ethereum.functions.ts:1]: Previously, this function was living in the ``useEthereum()`` hook, which was removed. This function in the future, should move into the ``dlc-btc-lib`` somehow. @scolear

File: src/app/functions/ethereum-account.functions.ts (hunk 1 above)
Comment (Polybius93) [reply src/app/functions/ethereum-account.functions.ts:21]: These functions were previously part of the ``useEthereumAccount()`` hook, but that was removed, so it's content were moved here.

File: src/app/functions/configuration.functions.ts
Comment (Polybius93) [inline src/app/functions/configuration.functions.ts:1]: These functions are new additions, these are the functions that can retrieve the deployment plans from the global variable ``appConfiguration`` and use it to return ``Contract`` objects with different parameters.

File: src/app/components/vaults-list/components/vaults-list-group-container.tsx
Comment (Polybius93) [inline src/app/components/vaults-list/components/vaults-list-group-container.tsx:1]: The ``recommendTokenToMetamask()`` function is no longer part of the ``useEthereumAccount()`` hook; instead, it is now a standalone function. As a result, both the import statement and its arguments have been updated accordingly.

File: src/app/components/proof-of-reserve/components/merchant-details/merchant-details.tsx
Comment (Polybius93) [inline src/app/components/proof-of-reserve/components/merchant-details/merchant-details.tsx:1]: The `fetchMintBurnEvents` function is no longer part of the `useEthereum` hook, as it has been removed and is now a standalone function. This change necessitates updates to the import statement and its arguments.

File: src/app/components/my-vaults/components/my-vaults-header/components/my-vaults-header-balance-info.tsx
Comment (Polybius93) [inline src/app/components/my-vaults/components/my-vaults-header/components/my-vaults-header-balance-info.tsx:1]: The function that returns the asset values no longer formats them; this task is now handled by the UI.

File: src/app/components/modals/select-wallet-modal/select-wallet-modal.tsx
Comment (Polybius93) [inline src/app/components/modals/select-wallet-modal/select-wallet-modal.tsx:1]: The login dispatch is no longer managed within the `useEthereumAccount()` hook, as it no longer exists. Instead, it is now handled directly within this component.

File: src/app/hooks/use-proof-of-reserve.ts [hunk 2]
```diff
@@ -1,76 +1,47 @@
...
-
 interface UseProofOfReserveReturnType {
   proofOfReserve: [number | undefined, MerchantProofOfReserve[]] | undefined;
 }
 
 export function useProofOfReserve(): UseProofOfReserveReturnType {
-  const { retrieveAllVaults, getAttestorGroupPublicKey } = useEthereum();
-  const { network } = useSelector((state: RootState) => state.account);
-
-  const [proofOfReserveHandler, setProofOfReserveHandler] = useState<
```
Comment (Polybius93) [inline src/app/hooks/use-proof-of-reserve.ts:23]: We are no longer maintaining this ``useState`` value since we now call it directly. This approach helps us avoid rerendering errors and the need for ``isNil`` checks.

File: src/app/hooks/use-vaults.ts [hunk 3]
```diff
@@ -19,67 +20,84 @@ interface UseVaultsReturnType {
...
+      });
+    setIsLoading(false);
   };
 
   useEffect(() => {
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    fetchVaultsIfReady();
+    ethereumUserAddress && void fetchVaultsIfReady(ethereumUserAddress, ethereumNetwork);
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [contractsLoaded]);
```
Comment (Polybius93) [inline src/app/hooks/use-vaults.ts:42]: We are no longer dependent on the `contractsLoaded` value, as the required `Contract` is now instantiated directly when the function is called (when the user address is set), rather than being a variable from another context.

File: src/app/providers/ethereum-network-configuration.provider.tsx
Comment (Polybius93) [inline src/app/providers/ethereum-network-configuration.provider.tsx:1]: This ``Context`` has become the backbone for all *Ethereum-related* actions and information. It prepares the required contracts for instantiation, depending on the currently selected vault.

File: src/app/store/slices/account/account.slice.ts [hunk 4]
```diff
@@ -1,33 +1,21 @@
-import { EthereumNetwork, ethereumArbSepolia, ethereumArbitrum } from '@models/ethereum-network';
+import { getEthereumNetworkByID } from '@functions/configuration.functions';
 import { WalletType } from '@models/wallet';
 import { createSlice } from '@reduxjs/toolkit';
+import { EthereumNetwork } from 'dlc-btc-lib/models';
 
 interface AccountState {
   address: string | undefined;
-  walletType: WalletType | undefined;
+  walletType: WalletType;
```
Comment (Polybius93) [inline src/app/store/slices/account/account.slice.ts:8]: To avoid nil checking, we set a default value instead of allowing this to be possibly undefined.

File: src/app/providers/balance-context-provider.tsx [hunk 5]
```diff
@@ -17,31 +22,38 @@ export const BalanceContext = createContext<VaultContextType>({
...
     }
   };
 
   useEffect(() => {
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    fetchBalancesIfReady();
+    ethereumUserAddress && void fetchBalancesIfReady(ethereumUserAddress);
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [contractsLoaded, fundedVaults]);
+  }, [ethereumUserAddress]);
```
Comment (Polybius93) [inline src/app/providers/balance-context-provider.tsx:56]: Add back fundedVaults dependency

File: src/app/store/slices/account/account.slice.ts [hunk 6]
```diff
@@ -45,10 +33,10 @@ export const accountSlice = createSlice({
     },
     logout: state => {
       state.address = undefined;
-      state.walletType = undefined;
+      state.walletType = WalletType.Metamask;
       state.dlcBTCBalance = 0;
       state.lockedBTCBalance = 0;
-      state.network = ethereumNetwork;
+      state.network = getEthereumNetworkByID(appConfiguration.enabledEthereumNetworkIDs.at(0)!);
```
Comment (Polybius93) [inline src/app/store/slices/account/account.slice.ts:39]: Make this a variable

**Review by sosaucily — APPROVED** [review]
LGTM, good job pushing through and accepting the feedback and making (what are hopefully) upgrades.
